include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -223,10 +255,21 @@ class fm_index
      * No guarantees.
      */
     void construct(text_t const & text)
+        //!\cond
+        requires !is_collection
+        //!\endcond
     {
          // text must not be empty
-        if (text.begin() == text.end())
+        if (std::ranges::begin(text) == std::ranges::end(text))
             throw std::invalid_argument("The text that is indexed cannot be empty.");
+
+        // Since we will later calculate rank + 1, the rank 255 + 1 will overflow to 0 which is reserved by sdsl
+        assert(!std::ranges::any_of(text,
+                                    [] (char_type r)
+                                    {
+                                        return to_rank(r) == static_cast<underlying_rank_t<char_type>>(255);
+                                    }));
```

Comment:
ah, I see, here's the check. But the cast is unsafe again, see the previous comment regarding rank_t.